plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -49,40 +59,75 @@ class FeatureDevSessionContext(val project: Project) {
     private val gitIgnoreFile = File(projectRoot.path, ".gitignore")
 
     init {
-        ignorePatternsWithGitIgnore = ignorePatterns + parseGitIgnore().map { Regex(it) }
+        ignorePatternsWithGitIgnore = (ignorePatterns + parseGitIgnore().map { Regex(it) }).toList()
     }
 
     fun getProjectZip(): ZipCreationResult {
-        val zippedProject = runReadAction { zipFiles(projectRoot) }
+        val zippedProject = runBlocking { zipFiles(projectRoot) }
```

Comment:
`withBackgroundProgress` so the user can cancel?

Reply:
Cancel by clicking the `x` in status bar? Sure